### PR TITLE
Add opentelemetry observability for analyses

### PR DESCRIPTION
**Pull request by rhelmot**
Should be a no-op when not explicitly enabled. Example code for enabling:

```
from opentelemetry import trace
from opentelemetry.sdk.trace import TracerProvider
from opentelemetry.sdk.trace.export import (
    BatchSpanProcessor,
    ConsoleSpanExporter,
)

provider = TracerProvider()
processor = BatchSpanProcessor(ConsoleSpanExporter())
provider.add_span_processor(processor)

# Sets the global default tracer provider
trace.set_tracer_provider(provider)
```

### CODE REVIEW

File: angr/analyses/analysis.py [hunk 1]
```diff
@@ -186,7 +190,45 @@ def prep(
         show_progressbar: bool = False,
     ) -> type[A]:
         @functools.wraps(self._analysis_cls.__init__)
+        @t.start_as_current_span(self._analysis_cls.__name__)
         def wrapper(*args, **kwargs):
+            span = telemetry.get_current_span()
+            sig = cast(Signature, self.__call__.__func__.__signature__)
+            bound = sig.bind(None, *args, **kwargs)
```
Comment (notxvilka) [inline angr/analyses/analysis.py:197]: This line throws an exception `TypeError: missing `func` argument` if the timeout in RDA occurs. I reverted this commit and the error disappears, allowing other threads to proceed just fine.

**Review by notxvilka — COMMENTED** [review]
I am not familiar with your code, could you please consider checking it yourself?

File: angr/analyses/analysis.py (hunk 1 above)
Comment (rhelmot) [reply angr/analyses/analysis.py:197]: Please open a new issue with a testcase that reproduces this problem.

Comment (rhelmot) [reply angr/analyses/analysis.py:197]: Also a backtrace will be helpful...

**Pull request merged**